servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/DefaultAsyncContextProvider.java

```diff
@@ -34,15 +34,18 @@
 import java.util.function.Consumer;
 import java.util.function.Function;
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
 import static java.lang.ThreadLocal.withInitial;
 
 final class DefaultAsyncContextProvider implements AsyncContextProvider {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultAsyncContextProvider.class);
     private static final ThreadLocal<ContextMap> CONTEXT_THREAD_LOCAL =
             withInitial(DefaultAsyncContextProvider::newContextMap);
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultAsyncContextProvider.class);
+    private static final boolean NOT_IS_DEBUG_ENABLED = !LOGGER.isDebugEnabled();
```

Comment:
Alternative name for consideration in a follow-up: `NO_DEBUG_LOGGING`

Reply:
Done in https://github.com/apple/servicetalk/pull/3183/commits/edf142afc86415dbcee94025947c7d40074a262a